pkg/apiserver/rest/dualwriter_mode1.go

```diff
@@ -135,7 +139,7 @@ func (d *DualWriterMode1) Delete(ctx context.Context, name string, deleteValidat
 		d.recordStorageDuration(err != nil, mode1Str, options.Kind, method, startStorage)
 	}()
 
-	return res, async, nil
+	return res, async, err
```

Comment:
This is going to return errors both coming from Legancy and Storage. Is that what we want?

Reply:
isn't the error on unified storage scoped to the goroutine?

Reply:
Yes, the `err` in the goroutine is redefined so it's safe. But I would argue that it's a good practice to return `nil` even if we know that `err` is `nil` at this point _right now_, since in the future we may ignore some previous error that we just log, and then accidentally return an error when we were supposed to ignore it, which would cause a bug.